hal/src/nRF52840/deviceid_hal.cpp

```diff
@@ -137,5 +137,19 @@ int hal_get_device_secret(char* data, size_t size, void* reserved)
     }
     return HAL_DEVICE_SECRET_SIZE;
 }
+
+int hal_set_device_secret(char* data, size_t size, void* reserved) {
+    if (!data || size != HAL_DEVICE_SECRET_SIZE || !isPrintable(data, HAL_DEVICE_SECRET_SIZE)) {
```

Comment:
As per Andrey's comment, we could check `data` and when it is `nullptr` we then write zeros to DCT to clear the mobile secret.